patches/tModLoader/Terraria/ModLoader/ModEmoteBubble.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.GameContent.UI;
+using Terraria.GameContent.UI.Elements;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader
+{
+	public abstract class ModEmoteBubble : ModType<EmoteBubble, ModEmoteBubble>
+	{
+		/// <summary>
+		/// The file name of this emote's texture file in the mod loader's file space.
+		/// </summary>
+		public virtual string Texture => (GetType().Namespace + "." + Name).Replace('.', '/');
+
+		/// <summary>
+		/// The internal ID of this EmoteBubble.
+		/// </summary>
+		public int Type { get; internal set; }
+
+		/// <summary>
+		/// This is the <see cref="EmoteBubble"/> instance.
+		/// </summary>
+		public EmoteBubble EmoteBubble { get; internal set; }
+
+		/// <summary>
+		/// This is the translation that is used behind <see cref="DisplayName"/>. The translation will show up as the emote command.
+		/// </summary>
+		public ModTranslation EmoteName { get; internal set; }
+
+		/// <summary>
+		/// This is the name that will show up as the emote command.
+		/// </summary>
+		public string DisplayName => EmoteName.GetTranslation(Language.ActiveCulture);
+
+		public sealed override void SetupContent() {
+			ModContent.Request<Texture2D>(Texture);
+			SetStaticDefaults();
+
+			if (EmoteName.IsDefault())
+				EmoteName.SetDefault(Name.ToLower()); // It will be the lowercase name of the class by default.
```

Comment:
Not sure if this is a good idea, we want to encourage use of localisation files.

Reply:
This is consistent with all other content, I think it is fine to assume english name from classname